### ISSUE TITLE
Weapons: Implement guns

### ISSUE THREAD

**Issue by assofohdz** [assofohdz/Examples#69]
Implement guns with

- [x] A cost
- [x] A firing delay
- [x] A current level
- [x] A max level
- [ ] Damage per level

**Comment by assofohdz** [comment]
On hold in order to create a better arena/map/settings setup first

**Comment by assofohdz** [comment]
**Current state** (as of `infinity` branch):

- [x] **Cost** — `GunCost` component (`api/src/infinity/es/ship/weapons/GunCost.java`), seeded by `PrizeSystem.handleAcquireGun` from `CoreGameConstants.GUNCOST`.
- [x] **Firing delay** — `GunFireDelay`, seeded from `CoreGameConstants.GUNCOOLDOWN`.
- [x] **Current level** — `GunCurrentLevel(Guns enum)`; `Guns` is a level enum with `LEVEL_1..LEVEL_4`.
- [x] **Max level** — `GunMaxLevel`, seeded to `Guns.LEVEL_4`.
- [ ] **Damage per level** — no `GunDamage`/`DamagePerLevel` component or implementation in the codebase. `WeaponsSystem` does damage application but currently uses a single `Damage` component on the projectile entity, not a per-level table on the firer.

**Probably what's blocked on:** picking the data model for damage-per-level. Two options:
- Per-firer component (`GunDamagePerLevel(int[] orInt)` on the ship) — read by `WeaponsSystem` when constructing the projectile's `Damage`.
- Per-arena `ShipConfig` field (extends Pattern 4's `ShipConfig` to nest `weapons.gun.damagePerLevel`), projected onto the firer's components at spawn — matches the per-ship-config-dictionary roadmap in [`.claude/ship-config-dictionary.md`](.claude/ship-config-dictionary.md) (the canonical SVS INI key `BulletFireEnergy` is already in the "pending port" table).

Either way, the `blocked` label can probably come off once a direction is picked — it's not blocked on infrastructure, just on a data-model decision.

**Comment by assofohdz** [comment]
Migrated to local-markdown tracker: `.scratch/guns/PRD.md`. This repo's canonical issue tracker is now local files; GitHub issues archived for historical reference.
